src/placeholder_file.rs

```diff
@@ -18,13 +21,26 @@ use crate::usn::Usn;
 
 // TODO: this struct could probably have a better name to represent files/dirs
 /// A builder for creating new placeholder files/directories.
+#[repr(C)]
 #[derive(Debug)]
 pub struct PlaceholderFile<'a>(CF_PLACEHOLDER_CREATE_INFO, PhantomData<&'a ()>);
 
 impl<'a> PlaceholderFile<'a> {
     /// Creates a new [PlaceholderFile][crate::PlaceholderFile].
-    pub fn new() -> Self {
-        Self::default()
+    pub fn new(relative_path: impl AsRef<Path>) -> Self {
+        Self(
+            CF_PLACEHOLDER_CREATE_INFO {
+                RelativeFileName: PCWSTR(
+                    U16CString::from_os_str(relative_path.as_ref())
+                        .unwrap()
+                        .into_raw(),
+                ),
+                Flags: CloudFilters::CF_PLACEHOLDER_CREATE_FLAG_NONE,
+                Result: Foundation::S_OK,
+                ..Default::default()
+            },
+            PhantomData,
+        )
```

Comment:
What's the reasoning for introducing a `relative_path` parameter to the constructor?

Reply:
RelativeFileName is a required parameter for pass_with_placeholder.